spec/user_spec.rb

```diff
@@ -0,0 +1,24 @@
+describe User do
+  describe '.create' do
+    it 'creates a new user' do
+      user = User.create(peeper: 'Lupita', username: 'Lupi', email: 'lupitareyes@example.es', password: 'lupitrona')
+
+      expect(user.id).to eq 3
+    end
+
+    it 'secures the password with BCrypt' do
+      pass = BCrypt::Password.create('lumber')
+
+      u = User.new(peeper: 'Lumberman', username: 'Lumbi', email: 'lumberman@example.es', password: pass)
+
+      expect(u.save).to eq true
+      expect(User.get(u.id).password).to eq pass
```

Comment:
Given that you're passing an encrypted password to `User#new` in the controller, this is more of an integration test, than part of the `user_spec`.  If, however, you were to move encryption into the `User` class this test would stay here (implemented slightly differently).

Reply:
Hey Eddie! Thanks for this comment, I would like to review this point with you as well.

I have implemented the encryption within the model (User class), like this:

```
class User
  include DataMapper::Resource
  validates_uniqueness_of :email, :username

  has n, :peeps

  property :id,         Serial
  property :peeper,     String
  property :username,   String
  property :email,      String
  property :password,   String, :length => 60

  before :save do
    BCrypt::Password.create(password)
  end
end
```

Now my question is, in the mentioned test I was checking whether or not after encrypting the password it was still the same in the expectation. Now, how should I test a dependancy as a Unit Test being BCrypt the dependancy? The only refactoring I can think of is just checking the same, but removing the BCrypt part, like:

```
    it 'secures the password with BCrypt' do
      u = User.new(peeper: 'Lumberman', username: 'Lumbi', email: 'lumberman@example.es', password: 'lumber')

      expect(u.save).to eq true
      expect(User.get(u.id).password).to eq 'lumber'
    end
```

This is one of the points I am more confused about my tests I believe. Thanks again in advance!